debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -146,24 +147,50 @@ protected synchronized void commit()  {
     @Override
     protected synchronized void stop() {
         LoggingContext.PreviousContext previousContext = taskContext.configureLoggingContext(CONTEXT_NAME);
+
         try {
             if (replicationStream.get() == null) {
                 logger.debug("already stopped....");
                 return;
             }
+
+            closeConnections();
+        } finally {
+            replicationStream.set(null);
+            executorService.shutdownNow();
+            previousContext.restore();
+        }
+    }
+
+    private void closeConnections() {
+        Exception closingException = null;
+
+        try {
             if (replicationConnection != null) {
                 logger.debug("stopping streaming...");
                 //TODO author=Horia Chiorean date=08/11/2016 description=Ideally we'd close the stream, but it's not reliable atm (see javadoc)
                 //replicationStream.close();
                 // close the connection - this should also disconnect the current stream even if it's blocking
                 replicationConnection.close();
             }
-        } catch (Exception e) {
-            throw new ConnectException(e.getCause() != null ? e.getCause() : e);
-        } finally {
-            replicationStream.set(null);
-            executorService.shutdownNow();
-            previousContext.restore();
+        }
+        catch(Exception e) {
+            closingException = e;
```

Comment:
Nice defensive programming, a single question 
Is not `closingException` lost when `replicationConnection.close()` throws an Exception but `typeResolverConnection` is either `null` or its closing does not throw an exception?